### ISSUE TITLE
[Autofix error] Simple string cause autofix  error

### ISSUE THREAD

**Issue by qarmin** [astral-sh/ruff#3717]
Ruff  1bac206995043d7d533338b7f8b7db37f37bf43b

```
ruff . --fix
```

File content 
```
STRI2NG = "Invalid 򐕸harcer sub "
```
[4407894774PY_FILE_TEST_260750.py.zip](https://github.com/charliermarsh/ruff/files/11066997/4407894774PY_FILE_TEST_260750.py.zip)

fails with message
```
error: Autofix introduced a syntax error. Reverting all changes.

This indicates a bug in `ruff`. If you could open an issue at:

    https://github.com/charliermarsh/ruff/issues/new?title=%5BAutofix%20error%5D

...quoting the contents of `/home/rafal/a.py`, along with the `pyproject.toml` settings and executed command, we'd be very appreciative!

/home/rafal/a.py:1:1: INP001 File `/home/rafal/a.py` is part of an implicit namespace package. Add an `__init__.py`.
/home/rafal/a.py:1:1: D100 Missing docstring in public module
/home/rafal/a.py:1:35: PLE2512 Invalid unescaped character SUB, use "\x1A" instead

```

**Comment by charliermarsh** [comment]
Can you include the raw file? I can't repro by copying that content from GitHub.

**Comment by qarmin** [comment]
It is fixed in 0.0.285
